python-test/features/steps/control_plane_policies.py

```diff
@@ -42,6 +50,36 @@ def new_policy(context):
     check_policies(context)
 
 
+@step('agent have {amount_of_policies} policies applied to it')
+def list_policies_applied_to_an_agent(context, amount_of_policies):
+    agent = get_agent(context.token, context.agent['id'])
+    context.list_agent_policies = list(agent['last_hb_data']['policy_state'].keys())
+    assert_that(len(context.list_agent_policies), equal_to(int(amount_of_policies)), f'Amount of policies applied to '
+                                                                                     f'this agent failed with '
+                                                                                     f'{amount_of_policies} policies')
+
+
+@step('the container logs contains the message "{text_to_match}" referred to all applied policies within {'
+      'time_to_wait} seconds')
+def check_agent_log_for_policy(context, text_to_match, time_to_wait):
+    assert_that(sorted(context.list_agent_policies), equal_to(sorted(context.policies_created.keys())))
+    for policy_id in context.list_agent_policies:
+        time_waiting = 0
+        sleep_time = 0.5
+        timeout = int(time_to_wait)
+        text_found = False
+        while time_waiting < timeout:
+            logs = get_orb_agent_logs(context.container_id)
+            text_found = check_logs_contain_message_for_policy(logs, text_to_match, policy_id)
+            if text_found is True:
+                break
+            time.sleep(sleep_time)
+            time_waiting += sleep_time
```

Comment:
This method doesn't ensure that **all** applied policies show up in the log. It asserts if at least one does.